src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs

```diff
@@ -161,22 +161,36 @@ protected List<PSKeyVaultIdentityItem> ListVaults(string resourceGroupName, Hash
             return vaults;
         }
 
-        public virtual IEnumerable<PSKeyVaultIdentityItem> ListResources(Rest.Azure.OData.ODataQuery<GenericResourceFilter> filter = null, ulong first = ulong.MaxValue, ulong skip = ulong.MinValue)
+        public virtual IEnumerable<PSKeyVaultIdentityItem> ListResources(Rest.Azure.OData.ODataQuery<GenericResourceFilter> filter = null)
         {
             IResourceManagementClient armClient = ResourceClient;
-
-            return new GenericPageEnumerable<GenericResource>(() => armClient.Resources.List(filter), armClient.Resources.ListNext, first, skip).Select(r => new PSKeyVaultIdentityItem(r));
+            var response = armClient.Resources.List(filter);
```

Comment:
Refactor to avoid duplicated code

Reply:
Introduce a new method
`ListPagable(Func<type of response> getFirstPage, Func<string, type of response> getNextPage)`
Then you can simplify these two methods to call `ListPagable`

Reply:
Refactored without delegate function as the signature of Resources.List(filter) and ResourceGroup.ListResources(resourcegroupName, filter) is different.